adagrams/game.py

```diff
@@ -1,11 +1,145 @@
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    letter_pool_list = []
+    for letter in LETTER_POOL:
+        for freq in range(LETTER_POOL[letter]):
+            letter_pool_list.append(letter)
+    letter_bank = random.sample(letter_pool_list, 10)    
+    return letter_bank
 
 def uses_available_letters(word, letter_bank):
-    pass
+    ### determines whether letters used are from letter_bank
+    word_cap = word.upper()
+    for letter in word_cap:
+        if letter not in letter_bank:
+            return False
+    
+    ### determines whether letters are used in appropriate quantities
+    ### creates dict with letters as keys and quantities for letter_bank
+    letter_bank_num_dict = {}
+    letter_bank_set = set(letter_bank)
+    for item in letter_bank_set:
+        letter_bank_count = 0
+        for val in letter_bank:
+            if item == val:
+                letter_bank_count += 1
+        letter_bank_num_dict[item] = letter_bank_count
+    
+    ### creates dict with letters as keys and quantities for user word
+    word_set = set()
+    word_num_dict = {}
+    for letter in word_cap:
+        word_set.add(letter)
+    for item in word_set:
+        word_set_count = 0
+        for letter in word_cap:
+            if item == letter:
+                word_set_count += 1
+        word_num_dict[item] = word_set_count
+
+    ### determines whether quantity of letters in user guess exceeds
+    ### available quantity of letters in letter_bank, if so returns false
+    for key in letter_bank_num_dict:
+        if key in word_num_dict.keys():
+            if letter_bank_num_dict[key] < word_num_dict[key]:
+                return False
+    return True
```

Comment:
I know we discussed this functionality during the Adagrams Wrap-Up and you discussed the were interested in more concise solutions so here is one for you:
```python
def uses_available_letters(word, letter_bank):
    cap_word = word.upper()

    for letter in cap_word:
       if letter not in letter_bank: return False

       if cap_word.count(letter) > letter_bank.count(letter): return False
    
    return True
```